packages/core/bootstrap/src/lib/util.ts

```diff
@@ -228,3 +228,41 @@ export const permutator = (options: string[], delimiter?: string): string[] | st
   const join = (combos: string[][]) => combos.map((p) => p.join(delimiter))
   return typeof delimiter === 'string' ? join(output) : output
 }
+
+/**
+ * @description
+ * Check existing (non-undefined) value for its type.
+ *
+ * @url
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#real-world_usage
+ *
+ * @param obj The value to type check
+ * @param fullClass (Optional) Whether to use polyfill for checking null
+ *
+ * @returns String describing type of obj
+ */
+export function type(obj: any, fullClass?: boolean): string {
```

Comment:
type is a reserved keyword in TS-land, maybe `narrowObjectType` or `specificObjectType` would be better

Reply:
Changed it to `deepType` because it handles every type, not just objects.